mixnet/util/src/lib.rs

```diff
@@ -1,41 +1,37 @@
-use std::{collections::HashMap, net::SocketAddr, sync::Arc};
+use std::{net::SocketAddr, sync::Arc};
 
-use parking_lot::Mutex;
+use crossbeam_skiplist::SkipMap;
+use spin::RwLock;
 use tokio::net::TcpStream;
 
 #[derive(Clone)]
 pub struct ConnectionPool {
-    pool: Arc<Mutex<HashMap<SocketAddr, Arc<tokio::sync::Mutex<TcpStream>>>>>,
+    pool: Arc<SkipMap<SocketAddr, Arc<RwLock<TcpStream>>>>,
```

Comment:
> As we know, we do not have any data race in TcpStream, so this spin lock will not consume too many CPU cycles.

I think skiplist is a good choice, rather than tokio::sync::Mutex. But, I’m not sure with spinlock since network IO will sometimes takes some time.